src/Platform/Microsoft.Testing.Platform/OutputDevice/Console/NonAnsiTerminal.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Globalization;
+using System.Text;
+
+using Microsoft.Testing.Platform.Helpers;
+
+namespace Microsoft.Testing.Platform.OutputDevice.Console;
+
+/// <summary>
+/// Non-ANSI terminal that writes text using the standard Console.Foreground color capabilities to stay compatible with
+/// standard Windows command line, and other command lines that are not capable of ANSI, or when output is redirected.
+/// </summary>
+internal class NonAnsiTerminal : ITerminal
+{
+    private readonly IConsole _console;
+    private readonly ConsoleColor _defaultForegroundColor;
+    private readonly StringBuilder _stringBuilder = new();
+    private bool _isBatching;
+
+    public NonAnsiTerminal(IConsole console)
+    {
+        _console = console;
+        _defaultForegroundColor = _console.GetForegroundColor();
+    }
+
+    public int Width => _console.IsOutputRedirected ? int.MaxValue : _console.BufferWidth;
+
+    public int Height => _console.IsOutputRedirected ? int.MaxValue : _console.BufferHeight;
+
+    public void Append(char value)
+    {
+        if (_isBatching)
+        {
+            _stringBuilder.Append(value);
+        }
+        else
+        {
+            _console.Write(value.ToString());
```

Comment:
The system console has an overload with `char`, so it's better to add it to `IConsole` and to use it here instead of forcing call to `ToString()`.

Reply:
done